### DISCUSSION TITLE
Question regarding kiali and mesh visualization

### DISCUSSION THREAD

**Discussion by Jack200062** [kiali/kiali#6862] [Q&A]
Hi! Can please someone explain me whether i need Jaeger(or any other tracing tool) to visualise traffic flow? 
Currently I have a problem, that kiali graph shows nothing, though there is a sidecar injected and metrics present in prometheus.

**Comment by jmazzitelli** [comment]
(moving to Q&A discussion)

**Comment by jmazzitelli** [comment]
You do not need tracing to see the traffic graph. See the FAQ "Why is my graph empty": https://kiali.io/docs/faq/graph/#emptygraph

**Reply by Jack200062** [reply]
Hi @jmazzitelli  Actually I don't see **istio_requests_total** metric in my prometheus, though there is a scraping configured and metrics being sent. Any ideas why there is no **istio_requests_total** metric?

**Reply by jmazzitelli** [reply]
I'm not sure what could be the cause of not having any `istio_requests_total` data in Prometheus (other than the obvious - there is no traffic flowing through your mesh) - but for sure without that metric data in Prometheus, your Kiali graph will be empty. So that explains the empty Kiali graph.

Assuming you have actual traffic flowing through your mesh, it sounds like Prometheus isn't truly scraping the endpoints or it is encountering a failure when doing so. Check your Istio configuration (Telemetry objects for example) to make sure Istio is configured correctly. And check that Prometheus isn't encountering errors during scrape attempts.. also confirm the scrape endpoints are valid.

Other than those recommendations, I can't think of what to do. At this point, it is outside the realm of Kiali. Kiali just cares about seeing istio_requests_total in Prometheus - how Prometheus gets and stores that data is an issue that Kiali doesn't deal with.

**Reply by Jack200062** [reply]
Okay, I see. I'll try to check all of the things you mentioned. Thanks!)
